### PR TITLE
Api auth

### LINKED ISSUE #34: User confirmation by appi
# REQUIREMENTS
It is required to add the user confirmation by devise token through the mobile application

# DESIGN
no apply

# Acceptance Criteria
List all required acceptance criteria
- [ ] validation user appi

### PR DESCRIPTION
**Pull request by ramasoftdev**
# Description
What did you implemented/Why did you implemented this:
 - Added and implement JWT for user session and validation from movil app.
 - Added a confirmation controller by API. For a user confirmation through movil app
 - Refactored set permission code in Role.rb
 - Fix HomeController and ReportersController bug running task for permission in app 
  
 # Testing
 Taking special check out at:
- app/controllers/api/v1/api_controller.rb
-app/controllers/api/v1/confirmations_controller.rb
- app/controllers/api/v1/sessions_controller.rb 
- app/controllers/api/v1/users_controller.rb
- config/routes.rb 
- lib/tasks/permission.rake

Missing some specs tests.

### CODE REVIEW

**Review by FranckZnt19 — APPROVED** [review]
excellent work, very clean!

**Review by mariormt17 — APPROVED** [review]
Good job my friend

File: alerta-covid/app/controllers/api/v1/api_controller.rb [hunk 1]
```diff
@@ -1,3 +1,39 @@
...
+
+  def decoded_token
+    if auth_header
+      token = auth_header.split('Bearer ')[1]
+      begin
+        JWT.decode(token, Rails.application.credentials.jwt_secret, true, algorithm: 'HS256', exp_leeway: 30)
+      rescue JWT::DecodeError, JWT::ExpiredSignature, JWT::VerificationError
+        nil
+      end
+    end
```
Comment (ArielMD) [inline alerta-covid/app/controllers/api/v1/api_controller.rb:22]: Nice

**Review by ArielMD — APPROVED** [review]
good job, now we only need to apply the styles to the views

**Pull request merged**